File: image-train-filters-app/src/main/java/org/jesperancinha/itf/android/main/MainFragmentManager.java
package org.jesperancinha.itf.android.main;

import android.view.View;

import org.jesperancinha.itf.android.config.ControlConfiguration;
import org.jesperancinha.itf.android.config.EditConfiguration;
import org.jesperancinha.itf.android.config.ImageConfiguration;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import static java.lang.Integer.parseInt;

@Builder
@AllArgsConstructor
@Getter
public class MainFragmentManager {

    private final MainFragment mainFragment;


    public String getOutputFileName() {
        return getEditConfiguration().getEditOutputFileName().getText().toString();
    }

    public String getDestinationImagePath() {
        return new File(mainFragment.getImageConfiguration().getCurrentSelectedFolder().getFile(), getOutputFileName()).getAbsolutePath();
    }

    public FileInputStream getImageFullStream() throws FileNotFoundException {
        return new FileInputStream(
                new File(getImageConfiguration().getCurrentSelectedFile().getFile().getAbsolutePath()));
    }

    public Character.UnicodeBlock getBlock() {
        return Character.UnicodeBlock.forName(getControlConfiguration().getSpiLanguageCode().getSelectedItem().toString());
    }

    public View getMainView() {
        return mainFragment.getView();
    }

    public int getFontSize() {
        return parseInt(getEditConfiguration().getEditFontSize().getText().toString());
    }

    public String getFontName() {
        return getControlConfiguration().getSpiFontType().getSelectedItem().toString();
    }

    public int getRangePercentage() {
        return parseInt(getEditConfiguration().getRange().getText().toString());
    }

    public int getDensity() {
        return parseInt(getEditConfiguration().getDensity().getText().toString());
    }

    public int getBackground() {
        return getControlConfiguration().getSvSelectedColor().getColor();
    }

    public EditConfiguration getEditConfiguration() {
        return this.mainFragment.getEditConfiguration();
    }

    public ControlConfiguration getControlConfiguration() {
        return this.mainFragment.getControlConfiguration();
    }

    public ImageConfiguration getImageConfiguration() {
        return this.mainFragment.getImageConfiguration();
    }

}

File: app-g2/app/src/main/java/org/jesperancinha/itf/app/MainFragmentManager.kt
package org.jesperancinha.itf.app

import android.graphics.Bitmap
import android.graphics.Typeface
import android.view.View
import lombok.Getter
import java.io.File
import java.io.FileInputStream
import java.io.IOException

class MainFragmentManager(protected var mainFragment: MainFragment) :
    MainFragmentManMainFragmentManagerConfigurationagerConfiguration() {
    val mainView: View?
        get() = mainFragment.view
    val outputFileName: String
        get() = getEditConfiguration().getEditOutputFileName().getText().toString()

    @Throws(IOException::class)
    protected fun setUpManager(): ChartizateManager<Int, Typeface, Bitmap> {
        return ChartizateManagerBuilderImpl()
            .backgroundColor(getControlConfiguration().getSvSelectedColor().getColor())
            .densityPercentage(getEditConfiguration().getDensity().getText().toString().toInt())
            .rangePercentage(getEditConfiguration().getRange().getText().toString().toInt())
            .distributionType(ChartizateDistributionType.Linear)
            .fontName(getControlConfiguration().getSpiFontType().getSelectedItem().toString())
            .fontSize(getEditConfiguration().getEditFontSize().getText().toString().toInt())
            .block(
                Character.UnicodeBlock.forName(
                    getControlConfiguration().getSpiLanguageCode().getSelectedItem().toString()
                )
            )
            .imageFullStream(
                FileInputStream(
                    File(
                        getImageConfiguration().getCurrentSelectedFile().getFile().getAbsolutePath()
                    )
                )
            )
            .destinationImagePath(
                File(
                    getImageConfiguration().getCurrentSelectedFolder().getFile(),
                    outputFileName
                ).getAbsolutePath()
            )
            .build()
    }
}